### ISSUE TITLE
Cannot read property 'addAsset' of undefined

### ISSUE THREAD

**Issue by allemanfredi** [MetaMask/metamask-plugin-beta#145]
Hello!

I'm trying to develop a plugin for Metamask and after having received the following error:

```Cannot read property 'addAsset' of undefined```

I took a look at the code and i noticed that the instance of __`AssetsController`__  declared within __metamask-controller.js__, is declared after the __`PermissionsController`__ one, 
which accesses __`this.assetsController`__ that has not yet been instantiated. 

```js
this.permissionsController = new PermissionsController({
  setupProvider: this.setupProvider.bind(this),
  keyringController: this.keyringController,
  assetsController: this.assetsController,
  openPopup: opts.openPopup,
  closePopup: opts.closePopup,
  provider: this.provider,
  getApi: this.getPluginsApi.bind(this),
  notifyDomain: this.notifyConnections.bind(this),
  notifyAllDomains: this.notifyAllConnections.bind(this),
}, initState.PermissionsMetadata)

 .....

this.assetsController = new AssetsController({
  // TODO: Persist asset state?
  // For now handled by plugin persistence.
})
```

By declaring __`this.assetsController`__ before of __`this.permissionsController`__ it works!

**Comment by rekmarks** [comment]
@allemanfredi Thank you for bringing this to our attention. Clearly we broke the initialization order at some point! Fixed in: 022230f

**Comment by allemanfredi** [comment]
I would like to ask you another thing: is it currently possible to assign an image to a custom token? I followed your example but it doesn't seem to work... I looked at the code (__`token-list.js`__) and the map (__`assetImages`__) that should contain the reference to the image is empty.

```js
return h('div', tokens.concat(pluginTokens).map((tokenData) => {
  tokenData.image = assetImages[tokenData.address]
  if (tokenData.customViewUrl) {
    tokenData.onClick = this.showPluginToken.bind(this, tokenData)
  }
  return h(TokenCell, tokenData)
}))```
